### PR TITLE
feat: add MANIFEST.in and more help

### PR DESCRIPTION
**Pull request by henryiii**
Brought up on Slack (can't pull up details, Slack is down currently).

Adds MANIFEST.in help, along with a few more notes and some helpful links. `include_package_data` was missing, as well.

A new and exiting addition to Scikit-HEP related to all this has been in the works during the holidays, and should come along soon, but this should help in the meantime.

### CODE REVIEW

File: pages/developers/packaging.md [hunk 1]
```diff
@@ -149,7 +160,7 @@ picked up correctly; just make sure you install the package and access it from
 there.
 
 The one place where the pep518 requirements do not get picked up is when you
-manually run `setup.py`, such as when doing `python setup.py sdist`. If you
+manually run `setup.py`, such as when doing `python setup.py sdist`[^1]. If you
```
Comment (eduardo-rodrigues) [inline pages/developers/packaging.md:163]:
```suggestion
manually run `setup.py`, such as when doing `python setup.py sdist` [^1]. If you
```

File: pages/developers/packaging.md [hunk 2]
```diff
@@ -337,9 +354,58 @@ extras["all"] = sum(extras.values(), [])
...
+2. SDist is processed to make a wheel (same virtual environment).
+3. The wheel is installed.
+
+The wheel does _not_ contain `setup.*`, `pyproject.toml`, or other build code.
+It simply is a `.tar.gz` file that is named `.whl` and has a simple mapping of
+directories to installation paths and a generic metadata format. "Installing"
+really is just copying files around, and pip also pre-compiles some bytecode
+for you while doing so.
+
+If you don't have a manifest.in, the "legacy" build procedure will skip the
```
Comment (eduardo-rodrigues) [inline pages/developers/packaging.md:373]:
```suggestion
If you don't have a MANIFEST.in, the "legacy" build procedure will skip the
```

**Review by eduardo-rodrigues — APPROVED** [review]
Looks great. Just a few trivial suggestions.

**Comment by henryiii** [comment]
Thank you! Not sure how I missed `manifest.in`...

**Pull request merged**